token/client/src/token.rs

```diff
@@ -765,23 +768,39 @@ where
     }
 
     /// Close account into another
-    pub async fn close_account<S: Signer>(
+    pub async fn close_account<S: Signers>(
         &self,
         account: &Pubkey,
         destination: &Pubkey,
-        authority: &S,
+        authority: &Pubkey,
+        signing_keypairs: &S,
     ) -> TokenResult<T::Output> {
-        self.process_ixs(
-            &[instruction::close_account(
-                &self.program_id,
-                account,
-                destination,
-                &authority.pubkey(),
-                &[],
-            )?],
-            &[authority],
-        )
-        .await
+        // XXX i know this is stupid. but maybe it would feel less stupid buried in a helper function!
+        let signer_pubkeys = signing_keypairs.pubkeys();
+        let signer_pubkey_refs: &[&Pubkey] = &signer_pubkeys
+            .iter()
+            .map(Borrow::borrow)
+            .collect::<Vec<&Pubkey>>();
```

Comment:
nit: You can gather all the refs in a vec without the `map(Borrow::borrow)` bit and just do `signer_pubkeys.iter().collect::<Vec<_>>()`

I know it looks goofy, but it's the only way

Reply:
oooh thats cool